src/data/battle_moves.h

```diff
@@ -4545,7 +4558,7 @@ const struct BattleMove gBattleMoves[MOVES_COUNT_Z] =
         .priority = 0,
         .flags = FLAG_MAKES_CONTACT | FLAG_PROTECT_AFFECTED | FLAG_MIRROR_MOVE_AFFECTED,
         .split = SPLIT_PHYSICAL,
-        .argument = STATUS1_PARALYSIS,
+        .argument = MOVE_EFFECT_WITH_DATA(MOVE_EFFECT_REMOVE_STATUS | MOVE_EFFECT_CERTAIN, STATUS1_PARALYSIS),
```

Comment:
I think Edu's big complaint is that this user-facing syntax is nasty, and I have to say I agree.

Perhaps a cleaner syntax could be something like:
```diff
+#define EFFECT_HIT_REMOVE_STATUS(status) /* ... */

-.effect = EFFECT_SMELLING_SALTS,
-.argument = STATUS1_PARALYSIS,
+.effect = EFFECT_HIT_REMOVE_STATUS(STATUS1_PARALYSIS),
```
I personally am a big fan of thinking of the effects themselves as having arguments, and reflecting that in the syntax.

One way to implement that macro could be:
```c
// In BattleMove we'd replace 'u16 effect;' and 'u16 argument'; with
// 'struct MoveEffect effect;'.
struct MoveEffect {
    u16 effect;
    u16 argument;
};

#define EFFECT_HIT_REMOVE_STATUS(status) (struct MoveEffect) { EFFECT_HIT_REMOVE_STATUS, MOVE_EFFECT_WITH_DATA(MOVE_EFFECT_REMOVE_STATUS | MOVE_EFFECT_CERTAIN, (status)) }
```
I would also consider making `argument` be a `union` because I'm not a huge fan of the manual bit twiddling, but it's not a high priority for me as long as the user is unlikely to interact with it (both when defining new moves or defining new effects).

There are other ways to implement it, e.g. by hiding a second initializer inside the argument to `.effect = EFFECT_HIT_REMOVE_STATUS(...)`, as in:
```c
#define EFFECT_HIT_REMOVE_STATUS(status) EFFECT_HIT_REMOVE_STATUS, .argument = MOVE_EFFECT_WITH_DATA(MOVE_EFFECT_REMOVE_STATUS | MOVE_EFFECT_CERTAIN, (status))
```
But I'm listing that mostly for reference because I think this form should be avoided if at all possible. (pret uses it for trainer parties because it's not able to refactor things into structs, but we don't have that limitation).

Reply:
Hmmmmmmmm this is a tricky one

I absolutely get where you're coming from and I agree that it's not exactly clear what's happening here BUT I've limited the scope of this PR to not include refactoring `BattleMove` - that's what the other PR is for.

The macros I came up with are essentially just compensation for the lack of anonymous structs which I would have used (I wouldn't count as refactoring as I wouldn't have to change any existing moves).

It sounds like the simple solution is to merge the other PR first then come back to this?

Reply:
I asked this on Discord, but while commuting I feel like I've settled more on a recommendation than a question: can you reduce this PR into just the changes which are relevant to Sheer Force (i.e. the ones around `secondaryEffectChance`), and then open a new PR for the other changes? (Or perhaps vice-versa would be cleaner, as a lot of the comments here seem to relate to non-Sheer Force changes)